src/components/course/list/student/CourseList.vue

```diff
@@ -44,20 +42,20 @@
                 default: false,
             },
         },
-
-        setup(props: any) {
-            let busy = ref(false);
+        emits: ["updated"],
+        setup(props: any, { emit }: any) {
+            let isLoading = ref(false);
             let lecturers = ref([] as Lecturer[]);
             let courses = ref([] as Course[]);
             let admittedCourses = ref([] as any[]);
             let username = ref("");
 
             onBeforeMount(async () => {
-                busy.value = true;
+                isLoading.value = true;
                 await getUsername();
                 await getAdmittedCourses();
                 await getCourses();
```

Comment:
```suggestion
                const promises = [];
                promises.push(getUsername());
                promises.push(getCourses());
                await Promise.all(promises);
                await getAdmittedCourses();
```
Could be done in parallel (not testet that though) as getCourses does not need the username